lib/Interpreter/CIFactory.cpp

```diff
@@ -338,9 +339,367 @@ static bool getVisualStudioDir(std::string &path) {
+    sysRoot.swap(SDKv);
+    return true;
+  }
+
+  return false;
+}
+
+static bool getISysRoot(std::string& sysRoot) {
+  using namespace llvm::sys;
+
+  // Some versions of OS X and Server have headers installed
+  if (fs::is_regular_file("/usr/include/stdlib.h"))
+    return false;
+
+  std::string SDKs("/Applications/Xcode.app/Contents/Developer");
+
+  // Is XCode installed where it usually is?
+  if (!fs::is_directory(SDKs)) {
+    // Nope, use xcode-select -p to get the path
+    if (FILE *pf = ::popen("xcode-select -p", "r")) {
+      SDKs.clear();
+      char buffer[512];
+      while (fgets(buffer, sizeof(buffer), pf) && buffer[0])
+        SDKs.append(buffer);
+
+      // remove trailing \n
+      while (!SDKs.empty() && SDKs.back() == '\n')
+        SDKs.resize(SDKs.size() - 1);
+      ::pclose(pf);
+    } else // Nothing more we can do
+      return false;
+  }
+
+  SDKs.append("/Platforms/MacOSX.platform/Developer/SDKs/");
+  if (!fs::is_directory(SDKs))
+    return false;
+
+
+  // Seems to make more sense to get the currently running SDK so any loaded
+  // libraries won't casue conflicts
+
+  // Try to get the SDK for whatever version of OS X is currently running
+  if (void *core = dlopen(
+          "/System/Library/Frameworks/CoreServices.framework/CoreServices",
+          RTLD_LAZY)) {
+    SInt32 majorVersion = -1, minorVersion = -1;
+    typedef ::OSErr (*GestaltProc)(::OSType, ::SInt32 *);
+    if (GestaltProc Gestalt = (GestaltProc)dlsym(core, "Gestalt")) {
+      Gestalt(gestaltSystemVersionMajor, &majorVersion);
+      Gestalt(gestaltSystemVersionMinor, &minorVersion);
+    }
+    ::dlclose(core);
+
+    if (majorVersion != -1 && minorVersion != -1) {
+      if (getISysRootVersion(SDKs, majorVersion, minorVersion, sysRoot))
+        return true;
+    }
+  }
+
+#define GET_ISYSROOT_VER(maj, min) \
+  if (getISysRootVersion(SDKs, maj, min, sysRoot)) \
+    return true;
+
+  // Try to get the SDK for whatever cling was compiled with
+  #if defined(MAC_OS_X_VERSION_10_11)
+    GET_ISYSROOT_VER(10, 11);
+  #elif defined(MAC_OS_X_VERSION_10_10)
+    GET_ISYSROOT_VER(10, 10);
+  #elif defined(MAC_OS_X_VERSION_10_9)
+    GET_ISYSROOT_VER(10, 9);
+  #elif defined(MAC_OS_X_VERSION_10_8)
+    GET_ISYSROOT_VER(10, 8);
+  #elif defined(MAC_OS_X_VERSION_10_7)
+    GET_ISYSROOT_VER(10, 7);
+  #elif defined(MAC_OS_X_VERSION_10_6)
+    GET_ISYSROOT_VER(10, 6);
+  #elif defined(MAC_OS_X_VERSION_10_5)
+    GET_ISYSROOT_VER(10, 5);
+  #elif defined(MAC_OS_X_VERSION_10_4)
+    GET_ISYSROOT_VER(10, 4);
+  #elif defined(MAC_OS_X_VERSION_10_3)
+    GET_ISYSROOT_VER(10, 3);
+  #elif defined(MAC_OS_X_VERSION_10_2)
+    GET_ISYSROOT_VER(10, 2);
+  #elif defined(MAC_OS_X_VERSION_10_1)
+    GET_ISYSROOT_VER(10, 1);
+  #else // MAC_OS_X_VERSION_10_0
+    GET_ISYSROOT_VER(10, 0);
+  #endif
+
+#undef GET_ISYSROOT_VER
+
+  // Scan the SDKs directory and use the latest
+  std::error_code ec;
+  std::vector<std::string> srtd;
```

Comment:
Can't we not allocate strings here but use llvm::StringRef/string_view. Just adding a //FIXME here is fine with me.